### ISSUE TITLE
FPS counter causes the game to crash on startup A LOT

### ISSUE THREAD

**Issue by ZackCDLVI** [Afr0Games/Project-Dollhouse#164]
![](http://i.imgur.com/IVW4Zkl.png)

**Comment by LetsRaceBwoi** [comment]
That's strange.  It works for me.

**Comment by Afr0** [comment]
That's... strange. Maybe it's a threading issue.

**Comment by LetsRaceBwoi** [comment]
Actually, this exact error happens to me from time to time; I was busy testing out @RHY3756547's buy mode creating a skill lot, but it does happen (albeit rarely) in the master branch, too.

**Comment by RHY3756547** [comment]
Always works for me...

**Comment by LetsRaceBwoi** [comment]
I am too :smile:

**Comment by ZackCDLVI** [comment]
It's periodic but, boy, when it does happen, it'll either stop after one restart or a few dozen.

**Comment by ZackCDLVI** [comment]
![](http://i.imgur.com/pGljZui.png)
For the purpose of this screenshot, I had it uncommented. Which is why you see the ``// Die!`` above. X)

**Comment by LetsRaceBwoi** [comment]
A simple fix, before the problem is fully resolved, would be to put it in a try loop which would display "FPS: ??" if it happens; but it depends on the frequency of the exception.  FPS should really be an integer, so I would use:
```c#
try
{
    SBatch.DrawString(m_SprFontBig, "FPS: " + ((int) FPS).ToString(), new Vector2(0,0), Color.Red);
}
catch (Exception e)
{
  if e.ToString().ToLower().Contains("text contains characters that cannot be resolved by this spritefont")
  {
    SBatch.DrawString(m_SprFontBig, "FPS: ??", new Vector2(0,0), Color.Red);
  }
}
```

**Comment by ZackCDLVI** [comment]
Issue has been fixed in merge #167.

**Comment by RHY3756547** [comment]
Yep, invalid characters don't render anymore. I was getting the crash with the McDonalds object when it tried to draw ™, which isn't baked into our spritefont.
